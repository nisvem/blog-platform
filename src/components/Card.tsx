import React from 'react';
import { Post } from '@/types';
import Link from 'next/link';

interface ICard extends React.HTMLAttributes<HTMLElement> {
  post: Post;
}

export const Card: React.FC<ICard> = ({ post, ...atr }) => {
  return (
    <Link href={`/blog/${post.id}`}

      className='bg-white rounded-xl text-dark-black p-4 flex flex-col h-full transition-all cursor-pointer shadow hover:bg-accent'
      {...atr}
    >
      <h2 className='text-xl font-bold mb-2'>{post.title}</h2>
      <p className='mb-auto'>{post.description.slice(0, 100)}...</p>
      <div className='flex flex-row gap-4 justify-between'>
      <p className='text-xs opacity-85 mt-2'>
        {new Date(post.createdAt).toLocaleDateString('en-EN', {
          day: 'numeric',
          month: 'long',
          year: 'numeric',
        })}
      </p>
      <p className='text-xs opacity-85 mt-2'>{post.author}</p>
      </div>

    </Link>
  );
};
