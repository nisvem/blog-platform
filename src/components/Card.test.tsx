import React from 'react';
import { render, screen } from '@testing-library/react';
import { Card } from './Card';
import '@testing-library/jest-dom';

const mockPost = {
  id: 1,
  title: 'Test Post',
  description: 'This is a test post',
  createdAt: new Date('2023-01-01T00:00:00Z'),
  author: 'Test Author',
};

describe('Card component', () => {
  it('renders with provided post data', () => {
    render(<Card post={mockPost} />);
    expect(screen.getAllByText(/Test Post/i)[0]).toBeInTheDocument();
    expect(screen.getByText(/This is a test post/i)).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Card post={mockPost} />);
    expect(asFragment()).toMatchSnapshot();
  });
}); 