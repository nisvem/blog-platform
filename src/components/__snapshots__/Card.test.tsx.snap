// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches the snapshot 1`] = `
<DocumentFragment>
  <a
    class="bg-white rounded-xl text-dark-black p-4 flex flex-col h-full transition-all cursor-pointer shadow hover:bg-accent"
    href="/blog/1"
  >
    <h2
      class="text-xl font-bold mb-2"
    >
      Test Post
    </h2>
    <p
      class="mb-auto"
    >
      This is a test post...
    </p>
    <div
      class="flex flex-row gap-4 justify-between"
    >
      <p
        class="text-xs opacity-85 mt-2"
      >
        January 1, 2023
      </p>
      <p
        class="text-xs opacity-85 mt-2"
      >
        Test Author
      </p>
    </div>
  </a>
</DocumentFragment>
`;
